### ISSUE TITLE
Add simple JSON rpc monitor

### ISSUE THREAD

**Issue by osnwt** [Optiminer/OptiminerZcash#11]
Would be great to have a listening TCP socket that receives connections in the form of stratum RPC and returns miner stats. Mandatory stats could be:

- miner version
- miner uptime in seconds
- pool connected
- solution rates, solutions found and rejected (stale) for each card
- the same as above but summary

It can be read by a simple monitor (like this: https://github.com/osnwt/ethmon). I used it with Claymore's miners and was happy to see miner stats from any web-enablde device. At the moment it requires parsing stdout/log files that is inconvenient. The monitor will add a feature of reading temperatures for fglrx and amdgpu-pro drivers and expose stats via web page.

I will port the monitor above for the miner as soon as it starts providing such statistics. Was going to add log parsing but don't like that approach.

**Comment by Optiminer** [comment]
Yeah, that's planned. Did not want to block the first release on that.

**Comment by wiryonolau** [comment]
also add gpu temperature if possible

**Comment by osnwt** [comment]
Currently have to use scripts to parse miner logs, fetch fan/temp values from fglrx/amdgpu-pro drivers and provide them via json-rpc to the web monitor. Still hope to see native interface and drop extra scripts.

![zecmon using shell script miner interface]
(http://i.imgur.com/nhXkKyV.png)

**Comment by Optiminer** [comment]
Except miner version, all of the data requested in the original report is now present as json on the monitoring port:

Example (It is not as nicely formatted by the miner as below but that should not matter for monitoring):
```
{
	"uptime": 84,
	"share": {
		"rejected": 0,
		"accepted": 34
	},
	"solution_rate": {
		"Total": {
			"60s": 334.316667,
			"1s": 353.000000
		},
		"GPU0": {
			"60s": 162.600000,
			"1s": 169.000000
		},
		"GPU1": {
			"60s": 171.850000,
			"1s": 181.000000
		}
	},
	"iteration_rate": {
		"Total": {
			"60s": 178.650000,
			"1s": 180.000000
		},
		"GPU0": {
			"60s": 86.616667,
			"1s": 88.000000
		},
		"GPU1": {
			"60s": 92.033333,
			"1s": 91.000000
		}
	},
	"stratum": {
		"target": "004189374bc6a7ef9db22d0e5604189374bc6a7ef9db22d0e5604189374bc6a7",
		"port": 3333,
		"host": "eu1-zcash.flypool.org",
		"connection_failures": 0,
		"connected": true
	}
}
```

**Comment by osnwt** [comment]
Excellent, thanks!

**Comment by Rinat056** [comment]
Can you share a modified version ethmon monitoring optiminer?

**Comment by osnwt** [comment]
Unfortunately, I waited for too long and ended up with some nodejs code that runs on every rig and collects statistics parsing optiminer log files and getting temps/fans using atitweak (fglrx) or /sys/class/hwmon  (amdgpu-pro). This app provides the same API as Claymore's miners, so I can use either miner or mix of them with a single ethmon port (rbrtio also ported it for zec here: https://github.com/rbrtio/zecmon).

But this nodejs code was done in a hackery style for internal use only, still there is no single version that could be published, and every rig has own hacks. Would be good to change original ethmon for this API, but it still needs to get temp/fan values (mandatory for me). Preparing for publishing takes time. It also seems that mining Equihash coins becomes senseless in terms of profit, so even less motivation to write anything for it at the moment. Prices dropped too much.

**Comment by bcmob** [comment]
[https://play.google.com/store/apps/details?id=ua.com.bcmob.claymoresdualminermonitor](url) Monitor for Android
